clients/src/main/java/org/apache/kafka/common/errors/ConfigurationMismatchException.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.errors;
+
+/**
+ * Replication factor for the topic is lower than min.insync.replicas. Messages are rejected when acks=all.
+ */
+public class ConfigurationMismatchException extends ApiException {
```

Comment:
Nit: Can the exception class name explicitly mention what configuration is mismatched? Something like `IncorrectReplicationFactorException | ReplicaConfigMismatchException` ?  The current javadoc and the class name don't seem to go hand in hand as the name suggests it can be reused for any kind of config mismatches.

Reply:
This seems beneficial indeed, agree.